std/array.d

```diff
@@ -3927,3 +3927,213 @@ unittest
     assert(appS.data == "hellow");
     assert(appA.data == "hellow");
 }
+
+/++
+Params: a = The input elements
+
+Returns: A static array constructed from `a`.
+
+The type of elements can be specified implicitly (`int[2] a = [1,2].asStatic;`)
+or explicitly (`float[2] a = [1,2].asStatic!float`). When `a` is a range,
```

Comment:
I think we shouldn't use `int[2] a` here as the entire point of asStatic was to be able to use `auto`. Maybe just say that the type is implicitly inferred to `int[2]` and leave the detailed explanation to the unittests?

Reply:
done